app/channel.go

```diff
@@ -1311,8 +1306,8 @@ func (a *App) ViewChannel(view *model.ChannelView, userId string, clearPushNotif
 
 	uchan := a.Srv.Store.Channel().UpdateLastViewedAt(channelIds, userId)
 
-	if pchan != nil {
-		if result := <-pchan; result.Err != nil {
+	if *a.Config().EmailSettings.SendPushNotifications && clearPushNotifications && len(view.ChannelId) > 0 {
+		if result := <-a.Srv.Store.User().GetUnreadCountForChannel(userId, view.ChannelId); result.Err != nil {
```

Comment:
We might want to run a load test on this, this is a high traffic bit of code and this is making this DB query happen more often

Reply:
@crspeller thoughts?

Reply:
Ya. This should be load tested.

Reply:
which is the proposal? we merge it, and pass a load test about it in master? or wait until we have it tested?

Reply:
Per chat on pre-release, we're going to test first then merge (correct me if I'm wrong guys)

Reply:
isn't this if racy? is unread count guaranteed to be updated only inside ViewChannel?
because I don't think it's correct and it might get cleared by other APIs(set to 0), and then this code would _never_ trigger as it seems in my testing

Reply:
I would need to review that with @enahum probably.

Reply:
There are 2 places where you can update the LastViewedAt attribute of the membership of the channel (the attribute which decide what you haven't readed). One of them, is when you explicitly use viewChannel from the api (basically when you try to mark the chanell as readed) and only when you are using a non-mobile application. The other place is when you add a member to a channel. This second case can be a problem, but is not a "common case". I going to investigate a bit more that case.

Reply:
Here were some corner cases related to other ways to send/clear notifications. I'm trying to change the approach to channel (`chan` golang type) approach to ensure the notifications send and cleaning order.